test/scenarios/index.ts

```diff
@@ -100,6 +100,38 @@ export const scenarios = {
       {},
     );
   },
+  /**
+   * Variation of the previous scenario in a nested workspace.
+   * 
+   */
+  dependentDoesNotMatchNestedWorkspaceVersion() {
+    return createScenario(
+      [
+        {
+          path: 'workspaces/a/packages/a/package.json',
+          before: wrapper('a', ['c@0.1.0']),
+          after: wrapper('a', ['c@0.0.1']),
+        },
+        {
+          path: 'workspaces/b/packages/b/package.json',
+          before: wrapper('b', [], ['c@0.2.0']),
+          after: wrapper('b', [], ['c@0.0.1']),
+        },
+        {
+          path: 'workspaces/b/packages/c/package.json',
+          before: wrapper('c', [], [], [], {
```

Comment:
What is the meaning of `[], [], []` here (and why for other wrappers there are less empty arrays)?

Reply:
They're entries that map to dependencies, devDependencies, and peerDependencies. An empty array just skips that particular type: https://github.com/JamieMason/syncpack/blob/master/test/mock.ts#L14. I'll probably change its API to an options object to make it clearer than it is now